docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+
+All sysl commands accept `--root` argument. Run `sysl -h` or `reljam -h` for more details.
+
+#### Multiple Declarations
+
+Sysl allows you to define an application in multiple places. There is no redefinition error in sysl.
+
+```
+UserService:
+  Login: ...
+
+UserService:
+  Register: ...
+```
+
+Result will be as-if it was declared like so:
+
+```
+UserService:
+  Login: ...
+  Register: ...
+```
+
+### Statements
+Our `MobileApp` does not have any detail yet on how it behaves. Let's use sysl statements to describe behaviour. Sysl supports following types of statements:
+  * [Text](#text)
+  * [Call](#Call)
+  * [Return](#return)
+  * [Control Statements](#control-statements)
+  * [Arguments](#arguments)
+
+#### Text
+Use simple text to describe behavior. See below for examples of text statements:
+```
+Server:
+  Login:
+    do input validation
+    "Use special characters like \n to break long text into multiple lines"
+    'Cannot use special characters in single quoted statements'
+```
+
+#### Call
+A standalone service that does not interact with anybody is not a very useful service. Use the `call` syntax to show interaction between two services.
+
+In the below example, MobileApp makes a call to backend Server which in turn calls database layer.
+
+```
+MobileApp:
+  Login:
+    Server <- Login
+
+Server:
+  Login:
+    do input validation
+    process data
+    "Use special characters like \n to break long text into multiple lines"
+    DB <- Save
+    return success or failure
+
+DB:
+  Save:
+    ...
+```
+##### You can generate the Sequence Diagram using the following command:
+
+```bash
+sysl sd -o 'call-login-sequence.png' -s 'MobileApp <- Login' call.sysl
+```
+You can omit the the `.sysl` and sysl will pickup the correct file.
+```bash
+sysl sd -o 'call-login-sequence.png' -s 'MobileApp <- Login' call
+```
+
+Here is the output that you should see:
+
+![](assets/call-Seq.png)
+
+See [assets/call.sysl](assets/call.sysl) for complete example.
+
+#### How sysl generates sequence diagram?
+Let's breakdown the `sd` aka `sequence diagram` command:
+```bash
+sysl sd -o 'call-login-sequence.png' -s 'MobileApp <- Login' call.sysl
+```
+  * `-o` specifies the output filename
+  * `-s` specifies the start endpoint
+  * `call.sysl` the source to start the analysis from
+
+Sysl analyzes the starting endpoint and finds all the `call`s that this endpoint makes to other endpoints (including the ones to other applications). It finds all the transitive dependencies till there are none.
+
+In the above diagram, `DB` is the last app in this flow. Sysl also captures the return data that each endpoint returns to its caller. See below for more details.
+
+#### Return response
+Express returning response to the caller by using the `return` keyword. See the example above on how to use the `return` keyword.
+```
+App:
+  Endpoint:
+    return STRING | TYPE
```

Comment:
i assume this isn't an example of valid syntax ( STRING | TYPE). it might be clearer to give an example of a string, and another example involving a type. or separately specify grammar so it doesnt look like an example